### PR TITLE
bind: Add types for options

### PR DESCRIPTION
**Pull request by Scriptkiddi**
###### Motivation for this change
fixes in the list of #76184

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/networking/bind.nix [hunk 1]
```diff
@@ -8,6 +8,35 @@ let
 
   bindUser = "named";
 
+  bindZoneOptions = {
+    name = mkOption {
+      type = types.str;
+      description = "Name of the zone.";
```
Comment (Lassulus) [inline nixos/modules/services/networking/bind.nix:14]: please remove trailing dots

Comment (aanderse) [reply nixos/modules/services/networking/bind.nix:14]: Option descriptions are supposed to end in a period. They aren't a short description and can consist of multiple sentences.

**Comment by petabyteboy** [comment]
This broke my eval. I will look into it.

**Comment by petabyteboy** [comment]
So I generate my zone files using nix-dns and it outputs a derivation (`types.package`) for the zone file.

```
error: A definition for option `services.bind.zones.[definition 1-entry 1].file' is not of type `string'. Definition values:
- In `/home/pbb/proj/nixfiles/modules/dns': <derivation /nix/store/mrrq2amn0sgwkapk661ak5gk71jvgjcf-0.c.a.4.f.0.a.2.ip6.arpa.zone.drv>
(use '--show-trace' to show detailed location information)
```

Should we change the `.file` option to `with types; oneOf [ str package ]` or do you think this is not a common use case?

**Comment by Scriptkiddi** [comment]
i think you should put "${}" in your config but i added the path option to allow rollbacks for zone files https://github.com/NixOS/nixpkgs/pull/111470/files

**Pull request merged**